server/src/main/java/org/jboss/as/server/SelfContainedContainer.java

```diff
@@ -0,0 +1,154 @@
+
+            Module.registerURLStreamHandlerFactoryModule(Module.getBootModuleLoader().loadModule(ModuleIdentifier.create("org.jboss.vfs")));
+            ServerEnvironment serverEnvironment = determineEnvironment( WildFlySecurityManager.getSystemPropertiesPrivileged(), WildFlySecurityManager.getSystemEnvironmentPrivileged(), ServerEnvironment.LaunchType.SELF_CONTAINED);
+            if (serverEnvironment == null) {
+                abort(null);
+            } else {
+                final Bootstrap bootstrap = Bootstrap.Factory.newInstance();
+                final Bootstrap.Configuration configuration = new Bootstrap.Configuration(serverEnvironment);
+                configuration.setConfigurationPersisterFactory(
+                        new Bootstrap.ConfigurationPersisterFactory() {
+                            @Override
+                            public ExtensibleConfigurationPersister createConfigurationPersister(ServerEnvironment serverEnvironment, ExecutorService executorService) {
+                                SelfContainedConfigurationPersister persister = new SelfContainedConfigurationPersister( containerDefinition );
+                                configuration.getExtensionRegistry().setWriterRegistry(persister);
+                                return persister;
+                            }
+                        });
+                configuration.setModuleLoader(Module.getBootModuleLoader());
+
+                bootstrap.bootstrap(configuration, Collections.<ServiceActivator>singletonList(new SelfContainedContentServiceActivator(content))).get();
+                return;
+            }
+        } catch (Throwable t) {
+            t.printStackTrace();
+            abort(t);
+        }
+    }
+
+    private static void abort(Throwable t) {
+        try {
+            if (t != null) {
+                t.printStackTrace(STDERR);
+            }
+        } finally {
+            SystemExiter.exit(ExitCodes.FAILED);
+        }
+    }
+
+    public static ServerEnvironment determineEnvironment(Properties systemProperties, Map<String, String> systemEnvironment, ServerEnvironment.LaunchType launchType) {
+        String serverConfig = null;
+        String initialServerConfig = null;
+        RunningMode runningMode = RunningMode.NORMAL;
+        ProductConfig productConfig;
+        if (serverConfig != null && initialServerConfig != null) {
+            throw ServerLogger.ROOT_LOGGER.cannotHaveBothInitialServerConfigAndServerConfig();
+        }
+        String hostControllerName = null; // No host controller unless in domain mode.
+        productConfig = new ProductConfig(Module.getBootModuleLoader(), WildFlySecurityManager.getPropertyPrivileged(ServerEnvironment.HOME_DIR, null), systemProperties);
+        return new ServerEnvironment(hostControllerName, systemProperties, systemEnvironment, serverConfig, null, launchType, runningMode, productConfig);
+    }
+
+    private static String fixPossibleIPv6URL(String val) {
```

Comment:
SelfContainedContainer is used by wildfly-boot, an external repository/project.

SelfContainedContainer itself needs to be in the server module, though, best I can tell, given classloaders.  I could try to move it outboard, though, I guess.